webvtt.go

```diff
@@ -357,29 +358,80 @@ func parseTextWebVTT(i string) (o Line) {
 			}
 
 		case html.TextToken:
-			if s := strings.TrimSpace(string(tr.Raw())); s != "" {
-				// Get style attribute
-				var sa *StyleAttributes
-				if len(webVTTTagStack) > 0 {
-					tags := make([]WebVTTTag, len(webVTTTagStack))
-					copy(tags, webVTTTagStack)
-					sa = &StyleAttributes{
-						WebVTTTags: tags,
-					}
-					sa.propagateWebVTTAttributes()
+			s := string(tr.Raw())
+			// Get style attribute
+			var sa *StyleAttributes
+			if len(webVTTTagStack) > 0 {
+				tags := make([]WebVTTTag, len(webVTTTagStack))
+				copy(tags, webVTTTagStack)
+				sa = &StyleAttributes{
+					WebVTTTags: tags,
 				}
-
-				// Append item
-				o.Items = append(o.Items, LineItem{
-					InlineStyle: sa,
-					Text:        unescapeWebVTT(s),
-				})
+				sa.propagateWebVTTAttributes()
 			}
+
+			// Append item
+			items := parseTextWebVTTText(sa, s)
+			o.Items = append(o.Items, items...)
 		}
 	}
 	return
 }
 
+func parseTextWebVTTText(sa *StyleAttributes, line string) []LineItem {
+	var ret []LineItem
+
+	// split the line by inline timestamps
+	indexes := inlineTimestampRegexp.FindAllStringSubmatchIndex(line, -1)
+	if len(indexes) > 0 {
+		// get the text before the first timestamp
+		s := strings.TrimSpace(line[:indexes[0][0]])
+		if s != "" {
+			ret = append(ret, LineItem{
+				InlineStyle: sa,
+				Text:        unescapeWebVTT(s),
+			})
+		}
+
+		for i, match := range indexes {
+			// get the text between the timestamps
+			endIndex := len(line)
+			if i+1 < len(indexes) {
+				endIndex = indexes[i+1][0]
+			}
+			s := strings.TrimSpace(line[match[1]:endIndex])
+			if s == "" {
+				continue
+			}
+
+			// get the timestamp
+			ts := line[match[2]:match[3]]
+
+			// Parse timestamp
+			t, err := parseDurationWebVTT(ts)
+			if err != nil {
+				log.Printf("astisub: parsing webvtt duration %s failed, ignoring: %v", ts, err)
+			}
+
+			ret = append(ret, LineItem{
+				InlineStyle: sa,
+				Text:        unescapeWebVTT(s),
+				StartAt:     t,
+			})
+		}
+	} else {
+		s := strings.TrimSpace(line)
```

Comment:
Since it moves to the top, could you simplify with
```go
if s := strings.TrimSpace(line); s != "" {
  return []LineItem{{
				InlineStyle: sa,
				Text:        unescapeWebVTT(s),
			}}
}
```